File: server/models/Match.ts
/** MANY TO ONE WITH TOURNAMENT */

import mongoose, { Schema } from 'mongoose';
import { IMatchModel } from '../../constants/interfaces';

const Match: Schema = new Schema({
  event: { type: String, required: true },
  round: { type: Number, required: true }, // round starting with 1
  fighter1: { type: String }, // username 1
  fighter2: { type: String }, // username 2
  winner: { type: String }, // username of winner
  uuid: { type: String, required: true, unique: true },
  tournamentId: { type: String, required: true }, // tournament uuid
});

export default mongoose.model<IMatchModel>('matches', Match);

File: HemaSite/Models/Match.cs
namespace HemaSite.Models
{
  public class Match
  {
    public int Id { get; set; }
    public int Round { get; set; }
    public string Fighter1 { get; set; }
    public string Fighter2 { get; set; }
    public string Event { get; set; }
    public Tournament Tournament { get; set; }
    public int TournamentId { get; set; }
    public string Winner { get; set; }
  }
}